src/lightning_lite/strategies/ddp_spawn.py

```diff
@@ -200,8 +196,7 @@ def _setup_distributed(self) -> None:
     def _get_process_group_backend(self) -> str:
         return self._process_group_backend or get_default_process_group_backend_for_device(self.root_device)
 
-    def _set_world_ranks(self, process_idx: int = 0) -> None:
-        self._local_rank = process_idx
+    def _set_world_ranks(self) -> None:
```

Comment:
This was a bug introduced in #14675 when applying #11073. The launcher already sets the local_rank, but here we were overwriting it to 0. A failing test was pointing me to it.